assembl/views/api2/auth.py

```diff
@@ -363,37 +363,24 @@ def reset_password(request):
     elif identifier:
         user, account = from_identifier(identifier)
         if not user:
-            if not discussion.preferences['generic_errors']:
-                raise JSONError(
-                    localizer.translate(_("This email does not exist")),
-                    code=HTTPNotFound.code)
-            else:
-                raise JSONError(
-                    localizer.translate(_(generic_error_message)),
-                    code=HTTPNotFound.code)
-                logger.error("This email does not exist.")
+            maybe_send_email(request, user, email, discussion=discussion)
         if account:
             email = account.email
     else:
         error = localizer.translate(_("Please give an identifier"))
         raise JSONError(error)
     if not email:
-        email = user.get_preferred_email()
+        if user:
+            email = user.get_preferred_email()
+        else:
+            email = None
         if not email:
-            if not discussion.preferences['generic_errors']:
-                error = localizer.translate(_("This user has no email"))
-            else:
-                error = localizer.translate(_(generic_error_message))
-                logger.error("This user has no email.")
-            raise JSONError(error, code=HTTPPreconditionFailed.code)
+            logger.error("This user has no email.")
```

Comment:
In the logs "This user" has absolutely no meaning. Put the identifier there so that we have tracing on the error.